### PR TITLE
return curried function in compose

### PR DESCRIPTION
**Pull request by stoeffel**
I'm new to Ramda but this should solve #736.

### CODE REVIEW

File: test/compose.js [hunk 1]
```diff
@@ -63,4 +63,10 @@ describe('compose', function() {
         assert.strictEqual(R.compose(f), f);
     });
 
+    it('should return curried function with same arity as it last argument', function() {
+        var concatAndJoin = R.compose(R.join(','), R.concat);
+        assert.strictEqual(concatAndJoin([1], [2]), '1,2');
+        assert.strictEqual(concatAndJoin.length, 2);
+        assert.strictEqual(concatAndJoin([1])([2]), '1,2');
+    });
```
Comment (davidchambers) [inline test/compose.js]: You should add a similar test case for `R.pipe`.

Comment (stoeffel) [inline test/compose.js]: I will do that.

File: src/internal/_createComposer.js [hunk 2]
```diff
@@ -16,7 +17,7 @@ module.exports = function _createComposer(composeFunction) {
                 while (idx--) {
                     fn = composeFunction(arguments[idx], fn);
                 }
-                return arity(length, fn);
+                return curryN(length, arity(length, fn));
```
Comment (CrossEye) [inline src/internal/_createComposer.js]: I don't think there's any reason for both `curryN` and `arity`.  You should be able to simply replace `arity` with `curryN` (untested).

**Comment by CrossEye** [comment]
We still need to decide [whether][co] before we take on [how][fi], I'm afraid.


  [co]: https://github.com/ramda/ramda/issues/736#issuecomment-70555440
  [fi]: https://github.com/ramda/ramda/pull/737/files

File: test/compose.js (hunk 1 above)
Comment (stoeffel) [inline test/compose.js]: Done.

**Comment by buzzdecafe** [comment]
*ping*

**Comment by buzzdecafe** [comment]
once #771 is in, i think it makes sense to do this as well

**Comment by stoeffel** [comment]
I will rebase as soon as #771 is in.

**Comment by buzzdecafe** [comment]
although #771 got whacked, i think we should go ahead with this, essentially returning `curryN(max(fns), f)` as we are doing elsewhere now,

**Comment by CrossEye** [comment]
I'm still nervous about this.  I agree that this should be the default behavior, but I really would like to find a way to offer a version of `compose` / `pipe` that don't do this in the case that the returned function has named optional parameters.

**Comment by buzzdecafe** [comment]
#985 is how `converge` works today. `compose` should have the same semantics IMO.

**Comment by stoeffel** [comment]
> #985 is how converge works today. compose should have the same semantics IMO.

I think so too.

I rebased this.

File: src/internal/_createComposer.js [hunk 3]
```diff
@@ -1,18 +1,22 @@
-var arity = require('../arity');
+var _noArgsException = require('./_noArgsException');
+var curryN = require('../curryN');
 
 
 /*
  * Returns a function that makes a multi-argument version of compose from
- * either _compose or _composeP.
+ * either _compose or _pCompose.
```
Comment (davidchambers) [inline src/internal/_createComposer.js]: This should be `_composeP` rather than `_pCompose`.

**Comment by davidchambers** [comment]
We've switched from using four spaces for indentation to two spaces. Please adjust the indentation accordingly. :)

**Comment by davidchambers** [comment]
Also, we no longer update __dist/ramda.js__ with each commit, so you should revert the changes to that file.

**Comment by stoeffel** [comment]
Yop, just saw that it will need more than just rebasing :smile: .

**Comment by stoeffel** [comment]
Now, it should be fin.

**Comment by davidchambers** [comment]
This looks good! It might be clearer to use an uncurried function in the test cases. As it stands one wonders whether `R.compose(R.join(','), R.concat)` evaluates to a curried function because `R.concat` is a curried function.

**Comment by buzzdecafe** [comment]
now the question is, do we need to have `composeV` (for variadic) with the old semantics to cover @CrossEye 's case?

otherwise, herd of :cow2:

**Comment by CrossEye** [comment]
I really hate having so many variants of this basic function.

But I think not offering that alternative does a big disservice to JS users who are used to working with functions with optional parameters, especially a final `options` argument.

**Comment by davidchambers** [comment]
Superseded by #1260

**Pull request closed without merging**